ProcessLib/HydroMechanics/HydroMechanicsFEM.h

```diff
@@ -0,0 +1,425 @@
+            ip_data._N_u = shape_matrices_u[ip].N;
+            ip_data._N_p = shape_matrices_p[ip].N;
+            ip_data._dNdx_p = shape_matrices_p[ip].dNdx;
+        }
+    }
+
+    void assemble(double const /*t*/, std::vector<double> const& /*local_x*/,
+                  std::vector<double>& /*local_M_data*/,
+                  std::vector<double>& /*local_K_data*/,
+                  std::vector<double>& /*local_rhs_data*/) override
+    {
+        OGS_FATAL(
+            "HydroMechanicsLocalAssembler: assembly without jacobian is not "
+            "implemented.");
+    }
+
+    void assembleWithJacobian(double const t,
+                              std::vector<double> const& local_x,
+                              std::vector<double> const& local_xdot,
+                              const double /*dxdot_dx*/, const double /*dx_dx*/,
+                              std::vector<double>& /*local_M_data*/,
+                              std::vector<double>& /*local_K_data*/,
+                              std::vector<double>& local_rhs_data,
+                              std::vector<double>& local_Jac_data) override
+    {
+        assert(local_x.size() == pressure_size + displacement_size);
+
+        auto p =
+            Eigen::Map<typename ShapeMatricesTypePressure::template VectorType<
+                pressure_size> const>(local_x.data() + pressure_index,
+                                      pressure_size);
+
+        auto u = Eigen::Map<typename ShapeMatricesType::template VectorType<
+            displacement_size> const>(local_x.data() + displacement_index,
+                                      displacement_size);
+
+        auto p_dot =
+            Eigen::Map<typename ShapeMatricesTypePressure::template VectorType<
+                pressure_size> const>(local_xdot.data() + pressure_index,
+                                      pressure_size);
+        auto u_dot = Eigen::Map<typename ShapeMatricesType::template VectorType<
+            displacement_size> const>(local_xdot.data() + displacement_index,
+                                      displacement_size);
+
+        auto local_Jac = MathLib::createZeroedMatrix<
+            typename ShapeMatricesType::template MatrixType<
+                displacement_size + pressure_size,
+                displacement_size + pressure_size>>(
+            local_Jac_data, displacement_size + pressure_size,
+            displacement_size + pressure_size);
+
+        auto local_rhs = MathLib::createZeroedVector<
+            typename ShapeMatricesType::template VectorType<displacement_size +
+                                                            pressure_size>>(
+            local_rhs_data, displacement_size + pressure_size);
+
+        typename ShapeMatricesTypePressure::NodalMatrixType laplace_p;
+        laplace_p.setZero(pressure_size, pressure_size);
+
+        typename ShapeMatricesTypePressure::NodalMatrixType storage_p;
+        storage_p.setZero(pressure_size, pressure_size);
```

Comment:
TBD;
In that particular matrix specific storage is used, no mass, so I'd like to keep it like it is.